underfs/swift/src/main/java/alluxio/underfs/swift/SwiftUnderFileSystem.java

```diff
@@ -261,7 +276,91 @@ public boolean mkdirs(String path, boolean createParent) throws IOException {
 
   @Override
   public boolean mkdirs(String path, MkdirsOptions options) throws IOException {
-    return true;
+    if (path == null) {
+      LOG.error("Attempting to create directory with a null path");
+      return false;
+    }
+    if (isDirectory(path)) {
+      return true;
+    }
+    if (isFile(path)) {
+      LOG.error("Cannot create directory {} because it is already a file.", path);
+      return false;
+    }
+    if (isRoot(path)) {
```

Comment:
isRoot is unnecessary as the call is part of isDirectory. removing the redundant call.